src/dbnode/persist/fs/msgpack/decoder.go

```diff
@@ -470,8 +469,12 @@ func (dec *Decoder) decodeIndexEntry(bytesPool pool.BytesPool) schema.IndexEntry
 	actualChecksum := dec.readerWithDigest.digest().Sum32()
 
 	// Decode checksum field originally added in V3
-	v := dec.decodeVarint()
-	indexEntry.IndexChecksum = v
+	indexEntry.IndexChecksum = dec.decodeVarint()
+	if dec.err != nil {
+		dec.err = fmt.Errorf("decode index entry encountered error: %s", dec.err)
```

Comment:
nit: `decode index entry error:` maybe?

Reply:
Wanted to make it consistent with the related `decode index entry encountered checksum mismatch` for easier discovery. Also wouldn't want to change the existing message (again, for search purposes) so I guess I'll stick to the current choice.